src/store/user.ts

```diff
@@ -0,0 +1,78 @@
+import { defineStore } from "pinia";
+import { UserService } from "@/services/UserService"
+import { DateTime } from 'luxon';
+import logger from '../logger';
+
+export const useUserStore = defineStore('user', {
+  state: () => {
+    return {
+      instanceUrl: '',
+      token: {
+        value: '',
+        expirationTime: ''
+      }as any
+    }
+  },
+  getters: {
+    getInstanceUrl(state) {
+      return state.instanceUrl
+    },
+    getBaseUrl(state) {
+      let baseURL = process.env.VUE_APP_BASE_URL;
+      if (!baseURL) baseURL = state.instanceUrl;
+      return baseURL.startsWith('http') ? baseURL : `https://${baseURL}.hotwax.io/nifi-api`;
+    },
+    getUserToken(state) {
+      return state.token;
+    },
+    isAuthenticated(state) {
+      let isTokenExpired = false
+      
+      if (state.token.expirationTime ) {
+        const currTime = DateTime.now().toMillis()
+        isTokenExpired = state.token.expirationTime < currTime
+      }
+      return state.token.value && !isTokenExpired
+    }
+  },
+  actions: {
+    async login( username: string, password:string ) {
+      try {
```

Comment:
Add loader to let user know that something is going on in the background for all actions wherever required.